pkg/endpoint/endpoint.go

```diff
@@ -2056,15 +2053,13 @@ func (e *Endpoint) identityLabelsChanged(owner Owner, myChangeRev int) error {
 	if e.identityResolutionIsObsolete(myChangeRev) {
 		e.Mutex.Unlock()
 
-		lastUse, err := identity.Release()
+		err := identity.Release()
 		if err != nil {
 			// non fatal error as keys will expire after lease expires but log it
 			elog.WithFields(logrus.Fields{logfields.Identity: identity.ID}).
 				WithError(err).Warn("Unable to release newly allocated identity again")
 		}
-		if lastUse {
-			owner.RemoveNetworkPolicy(e.SecurityIdentity.ID)
-		}
+		owner.RemoveNetworkPolicy(e.IPv6, e.IPv4)
```

Comment:
I don't think this call makes sense anymore. We should only call this when the endpoint is destroyed, which is not the case here.

Reply:
This is the case handling labels change on the endpoint. This could affect the policy, so I think we need to clean up the old NetworkPolicy.